gateway/fedimint-gateway-server/src/lib.rs

```diff
@@ -180,6 +181,30 @@ impl Display for GatewayState {
     }
 }
 
+/// Helper struct for storing the registration parameters for LNv1 for
+/// the HTTP and Iroh endpoints.
+#[derive(Debug, Clone)]
+struct GatewayRegistrationInfo {
+    pub api: SafeUrl,
+    pub id: PublicKey,
+}
+
+impl GatewayRegistrationInfo {
+    pub fn new_iroh(iroh_sk: &iroh::SecretKey) -> anyhow::Result<Self> {
+        // Compute a gateway id, which is a secp256k1 key, from the `iroh_sk`, which is
+        // a ed25519 key
+        let iroh_pk_hash = sha256::Hash::const_hash(iroh_sk.public().as_bytes());
+        let iroh_gw_id =
```

Comment:
I'm a bit confused here. We take a secret key, it's public key, then we hash it, then we use it a secret, to take it's public key again? Is that how it is supposed to be? It's not much of a secret if anyone can rederive it from public information, so I'm supa confused.

 And why do we need to consensus-encode something that is already fixed byte array?

Reply:
I will change this to pass in the public key.

The gateway id doesn't need to be secret, it just needs to be unique. I wanted something derived from the iroh key, but even that isn't a requirement. Let me know if there's a better idea.

> And why do we need to consensus-encode something that is already fixed byte array?

I probably missed a function to convert a `Hash` to ` &[u8]`